### PR TITLE
Hcp segmentation

### PR DESCRIPTION
**Pull request by claireguichon**

### CODE REVIEW

File: pynet/core.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def test(self, loader, with_logit=False, predict=False):
                         targets.append(item.to(self.device))
                 if len(targets) == 1:
                     targets = targets[0]
-                elif len(targers) == 0:
+                elif len(targets) == 0:
                     targets = None
```
Comment (AGrigis) [inline pynet/core.py]: Avoid conflicts: already fixed on master.

File: pynet/datasets/echocardiography.py [hunk 2]
```diff
@@ -80,7 +80,6 @@ def to_categorical(y, num_classes):
         if small:
             logger.info("Small mode, select only the 50 first images.")
             files = files[:50]
-        nb_files = len(files)
         data = []
```
Comment (AGrigis) [inline pynet/datasets/echocardiography.py:83]: Avoid conflicts: the small option has been removed from this fetcher.

File: pynet/datasets/hcp.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import os
+from collections import namedtuple
+
+import boto3
```
Comment (AGrigis) [inline pynet/datasets/hcp.py:4]: boto3 & nibabel are now dependencies of the package: add them in pynet/info.py & doc/requirements.txt.

File: pynet/datasets/hcp.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    masks.append(mask)
+
+    metadata["name"].append(subject[11:-1])
+    metadata["label"].append(label)
+
+    return data, mask, metadata
+
+
+def fetch_data(datasetdir, low=False, small=False):
+    """ Fetch/prepare the dataset for pynet.
```
Comment (AGrigis) [inline pynet/datasets/hcp.py:110]: Add space here.

File: pynet/datasets/hcp.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    Parameters
+    ----------
+    datasetdir: str
+        the dataset destination folder.
+    low: bool
+        set images in low resolution
+    small: bool
+        fetch 45 brains if true, else 1200 brains
```
Comment (AGrigis) [inline pynet/datasets/hcp.py:118]: Add space here.

File: pynet/datasets/hcp.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    img_data = np.append(img_data[2:-2, :, 2:-2], np.zeros((256, 9, 256)), axis=1)
+    if low:
+        img_data = img_data.reshape(32, 8, 40, 8, 32, 8).mean(axis=(1, 3, 5))
+
+    if binary:
+        img_data = 1 * (img_data > img_data.mean())
+    return img_data
+
+
+def extract_hcp_data(datasetdir, data, masks, s3, bucket, subject, label, low, metadata):
```
Comment (AGrigis) [inline pynet/datasets/hcp.py]: General remark we have to make this function as generic as possible. In the future we may use it to retrieve other data.
Simplify the use fct(outdir, s3, bucket, subject, modality, low) -> arr.
Add a docstring to describe the input parameters.
Function name get_hcp_data?
Label is not clear. I think we can replace this by modality.

File: pynet/datasets/hcp.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    if label == 'T1w':
+        obj = s3.Object(bucket, subject + label + '/T1w_acpc_dc.nii.gz')
+    else:
```
Comment (AGrigis) [inline pynet/datasets/hcp.py]: elif modality == "MNINonLinear":
...
else:
    raise ValueError(...)

File: pynet/datasets/hcp.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        obj = s3.Object(bucket, subject + label + '/T1w.nii.gz')
+
+    if not os.path.isdir(datasetdir + subject + label):
```
Comment (AGrigis) [inline pynet/datasets/hcp.py]: Please use os.path.join() in order to be linux and windows compliant, ie. don't hardcode the path separator /.
Take this remark into account for the whole code.

File: pynet/datasets/hcp.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        os.mkdir(datasetdir + subject + label)
```
Comment (AGrigis) [inline pynet/datasets/hcp.py]: You need to create two levels: os.makedirs?

File: pynet/datasets/hcp.py [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+import boto3
+import nibabel as nib
+import numpy as np
+import pandas as pd
+
+Item = namedtuple("Item", ["input_path", "output_path", "metadata_path"])
+
+
+def nii_to_np_arr(filename, low=False, binary=False):
+    img = nib.load(filename)
```
Comment (AGrigis) [inline pynet/datasets/hcp.py:28]: Function name: load or load_image.
Add docstring.
Explain that you want to create an array with shape 256 ^ 3 (not obvious from the code).

File: pynet/datasets/hcp.py [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    img_data = img.get_data()
+    img_data = np.append(img_data[2:-2, :, 2:-2], np.zeros((256, 9, 256)), axis=1)
+    if low:
+        img_data = img_data.reshape(32, 8, 40, 8, 32, 8).mean(axis=(1, 3, 5))
```
Comment (AGrigis) [inline pynet/datasets/hcp.py]: To downsample the image you may prefer using  scipy.ndimage.zoom or other dedicated functions.

File: pynet/datasets/hcp.py [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+    if binary:
+        img_data = 1 * (img_data > img_data.mean())
```
Comment (AGrigis) [inline pynet/datasets/hcp.py]: Strange. If you want to binarize the array you can do arr[arr > 0] = 1.

Comment (claireguichon) [reply pynet/datasets/hcp.py]: arr[arr > 0] = 1 makes the mask bigger in low resolution. That's why I used the mean. Using scipy.ndimage.zoom, values can also become greater than 1 or less than 0

**Pull request closed without merging**